<launch>
    <param name="robot_description" command="$(find xacro)/xacro $(find ur_description)/urdf/ur5e.xacro" />
    <param name="use_sim_time" value="true"/>

    <rosparam>
        MujocoPlugins:
          - {type: mujoco_ros_control/MujocoRosControlPlugin, hardware: {type: mujoco_ros_control/DefaultRobotHWSim, control_period: 0.002}}
          - {type: ibvs_ur5_sim/StereoPseudoDepthPlugin}
    </rosparam>
    <param name="left_camera_name" type="str" value="left_camera"/>
    <param name="right_camera_name" type="str" value="right_camera"/>
    <param name="aruco_marker_name" type="str" value="aruco_target"/>
    <param name="aruco_marker_size" type="double" value="0.10"/>
    <rosparam>
        cam_config:
            left_camera: {stream_type: 1, frequency: 30}
            right_camera: {stream_type: 1, frequency: 30}
    </rosparam>

    <node name="mujoco_server" pkg="mujoco_ros" type="mujoco_node" output="screen">
        <param name="modelfile" value="$(find ibvs_ur5_sim)/worlds/ur5e_ibvs_scene.xml" />
        <param name="headless" value="false"/>
    </node>

    <rosparam file="$(find ibvs_ur5_sim)/config/ur5_controllers.yaml"/>


    <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" />
    <node pkg="tf" type="static_transform_publisher" name="map_to_base_broadcaster" args="0 0 0 0 0 0 map base_link 100" />
    

    <node name="controller_spawner" pkg="controller_manager" type="spawner" output="screen"
          args="joint_state_controller scaled_pos_joint_traj_controller" />

    <node name="aruco_detector" pkg="ibvs_ur5_sim" type="aruco_detector_node.py" output="screen"/>
    <node name="initial_pose_setter" pkg="ibvs_ur5_sim" type="set_initial_pose.py" output="screen"/>

</launch>